tests/python_tests/ov_genai_test_utils.py

```diff
@@ -89,6 +88,7 @@ def read_model(params, **tokenizer_kwargs):
     from optimum.intel.openvino import OVModelForCausalLM
     from transformers import AutoTokenizer
     hf_tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)
+    hf_tokenizer.padding_side = tokenizer_kwargs.pop("padding_side", "right")
```

Comment:
why do we need to patch HF's Tokenizer padding side? 

OpenVINO Tokenizers always use `paddind_size=left`, so why is it not the same for HF?

Reply:
>OpenVINO Tokenizers always use paddind_size=left, so why is it not the same for HF?

By default we do not enforce any side during conversion. OV Tokenizers uses the padding side from hf_tokenizer object which can both right/left.

Reply:
> In CB we don't have padding, let's assume we have batch 2 with, in CB it will be
```
[1, 42, 98]
[1, 67, 89, 0, 56, 8]
```
> While in the vanilla LLMPipeline, since most of the models are right padded, in most cases we would have
```
[1, 42, 98, pad, pad, pad]
[1, 67, 89,     0,    56,    8]
```
> And results would not match for right padded models.

Shouldn't vanilla LLMPipeline suppress pad tokens with attention_mask?

Reply:
> Shouldn't vanilla LLMPipeline suppress pad tokens with attention_mask?

It can, but position IDs between tokens will different on an amount of # pad tokens.